pkg/asset/installconfig/azure/validation.go

```diff
@@ -88,59 +88,66 @@ func ValidateDiskEncryptionSet(client API, ic *types.InstallConfig) field.ErrorL
+			errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d GB Memory", req.minimumMemory)
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
+		}
+	} else {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, "capability not found: MemoryGB"))
 	}
 
-	ultraSSDAvailable := false
-	for _, capability := range *typeMeta.Capabilities {
+	val, ok = capabilities["PremiumIO"]
+	if diskType == "Premium_LRS" && ok {
+		if strings.EqualFold(val, "False") {
+			errMsg := fmt.Sprintf("PremiumIO not supported for instance type %s", instanceType)
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("osDisk", "diskType"), diskType, errMsg))
+		}
+	} else {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, "capability not found: PremiumIO"))
+	}
 
-		if strings.EqualFold(*capability.Name, "vCPUsAvailable") {
-			cpus, err := strconv.ParseFloat(*capability.Value, 0)
-			if err != nil {
-				return append(allErrs, field.InternalError(fieldPath, err))
-			}
-			if cpus < float64(req.minimumVCpus) {
-				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d vCPUsAvailable", req.minimumVCpus)
-				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
-			}
-		} else if strings.EqualFold(*capability.Name, "MemoryGB") {
-			memory, err := strconv.ParseFloat(*capability.Value, 0)
-			if err != nil {
-				return append(allErrs, field.InternalError(fieldPath, err))
-			}
-			if memory < float64(req.minimumMemory) {
-				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d GB Memory", req.minimumMemory)
-				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
-			}
-		} else if diskType == "Premium_LRS" && strings.EqualFold(*capability.Name, "PremiumIO") {
-			if strings.EqualFold(*capability.Value, "False") {
-				errMsg := fmt.Sprintf("PremiumIO not supported for instance type %s", instanceType)
-				allErrs = append(allErrs, field.Invalid(fieldPath.Child("osDisk", "diskType"), diskType, errMsg))
-			}
-		} else if strings.EqualFold(*capability.Name, "UltraSSDAvailable") {
-			ultraSSDAvailable = strings.EqualFold(*capability.Value, "True")
-		} else if strings.EqualFold(*capability.Name, string(aztypes.VMnetworkingTypeAccelerated)) {
-			if vmNetworkingType == string(aztypes.VMnetworkingTypeAccelerated) && !strings.EqualFold(*capability.Value, "True") {
+	if vmNetworkingType != "" {
+		val, ok = capabilities[string(aztypes.AcceleratedNetworkingEnabled)]
+		if ok {
+			if vmNetworkingType == string(aztypes.VMnetworkingTypeAccelerated) && !strings.EqualFold(val, "True") {
 				errMsg := fmt.Sprintf("vm networking type is not supported for instance type %s", instanceType)
 				allErrs = append(allErrs, field.Invalid(fieldPath.Child("vmNetworkingType"), vmNetworkingType, errMsg))
 			}
+		} else {
```

Comment:
if `vmNetworkingType == Basic` and `ok == false` this will throw an error, but I believe that is a valid state.

Reply:
I think we should take @r4f4's suggestion here and only validate if the vmNetworkingType is set to `Accelerated`. We don't need to validate basic as it is the default setting. I'll update.